Add function

leonard33 opened the pull request:
Add and delete functionality added

leonard33 committed:
store setup

leonard33 committed:
folder changed to src

leonard33 committed:
add and remove functionality

KDlamini commented on src/Components/Booklist.js:
```diff
@@ -1,32 +1,13 @@
-  ];
+  const books = useSelector((state) => state.bookReducer);
+  console.log(books, "initial state books");
+
```
Please remove the console statement on your component. It is messy to leave console statements in your code and can result in performance issues.

KDlamini commented on src/Components/Bookitem.js:
```diff
@@ -1,16 +1,29 @@
+
+  const deletehandler = () => {
+    dispatch(deletedata(book.id));
+  };
```
It is a great practice to use all props that are deconstructed in your component. In your case, you have two props that are deconstructed but never used.

- Since you deconstructed `id` in your `Bookitem` component, it would be best to utilize it on line 13 instead of `book.id` use `id` directly.
```
dispatch(deletedata(id));
```

- Similarly, you don't seem to be using the `book` prop in your component, therefore it is best that you do not deconstruct it. It is best to only deconstruct props that you will use.
- Kindly remove the commented code as well since you do not need it. It is best practice to follow the YAGNI principle. That includes removing the import of `import { useSelector } from "react-redux";`.
![image](https://user-images.githubusercontent.com/38088818/172639291-d24bd550-d749-4505-8f04-73877277bf6a.png)

KDlamini commented on src/redux/books/books.js:
```diff
@@ -0,0 +1,40 @@
+/*[
+  ...state.slice(0, action.index),
+  ...state.slice(action.index + 1, state.length)
+]; */
```
Also here, kindly remove commented code that you are not using. It is best practice to follow the YAGNI principle.

KDlamini commented on src/Components/Categories.js:
```diff
@@ -1,10 +1,17 @@
   return (
     <div className="prog">
-      <button type="button" className="btn-progress">
+      <button type="button" className="btn-progress onClick={statushandler} ">
```
The onClick event for checking status does not work. This is because you accidentally included your onClick inside the quotations for className.

- You have `className="btn-progress onClick={statushandler} "` instead of `className="btn-progress" onClick={statushandler}`
![image](https://user-images.githubusercontent.com/38088818/172640985-b3a26495-e465-4a02-87d8-90f8d16442f7.png)

- You might also need to fetch the state using `useSelector` and render the status on the page. When I click on the `Check Status` button, `Under Construction` should be displayed.

KDlamini commented on src/redux/configureStore.js:
```diff
@@ -0,0 +1,17 @@
+const rootReducer = combineReducers({
+  bookReducer: bookReducer,
+  // categoriesReducers: categoriesReducers,
+});
```
- Kindly uncomment the `categoriesReducers` so that you may be able to fetch the categories state in your Categories component to display on the page.

KDlamini reviewed (changes requested):
### Hi @leonard33 🙋‍♀️,  

Great work so far! 💯 
There are some issues that you still need to work on to improvee your project according to the requirements. You are almost there! 💪🏾 

![soclose](https://user-images.githubusercontent.com/38088818/172644716-d36f867b-5760-45d4-b315-07a6fa3a8609.gif)

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

leonard33 committed:
correct review

leonard33 commented:
correction PR
- Removed console log statements
- Removed all unused props that are deconstructed
- Removed commented codes
- onClick event for checking status corrected.
- uncommented categories reducer

leonard33 committed:
correct review

leonard33 commented:
-status working correctly

Kingstalux commented on src/Components/Categories.js:
```diff
@@ -6,16 +6,18 @@ import { checkstatus } from "../redux/categories/categories";
 
 const Categories = () => {
   const categories = useSelector((state) => state.categoriesReducer);
+  console.log(categories);
```
- [ ] (OPTIONAL) Kindly remove all console statements from your code so as to improve performance

Kingstalux commented on src/Components/Bookitem.js:
```diff
@@ -5,9 +5,8 @@ import { useDispatch } from "react-redux";
 // import { v4 as uuidv4 } from "uuid";
 
-const Bookitem = ({ id, title, author, genre, chapter, book }) => {
+const Bookitem = ({ title, author, genre, chapter, book }) => {
```
- [ ] (OPTIONAL) Kindly display both the title and author of a book. Right now only the author displays. Kindly fix this 👍 
- Your site:
![Annotation 2022-06-08 165659](https://user-images.githubusercontent.com/85113567/172663164-ac8d6d67-2211-4cac-bfb5-f1609dc080a9.jpg)

Kingstalux commented on src/Components/Bookitem.js:
```diff
@@ -5,9 +5,8 @@ import { useDispatch } from "react-redux";
 import { deletedata } from "../redux/books/books";
 // import { v4 as uuidv4 } from "uuid";
```
- [ ] (OPTIONAL) Kindly remove all commented code so as to keep your work clean 💯

Kingstalux reviewed (approved):
Hi @leonard33,
# Status: Approved ✔️  ✔️ ✔️ 

![68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966](https://user-images.githubusercontent.com/85113567/172663671-16879646-fd4e-42a9-8cb9-842cb1eaed41.gif)

## Highlights 👏 
✅ App has the correct functionality 🔥 

Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

_check below for optional suggestions_
- [ ] (OPTIONAL) Kindly provide a more detailed PR description which should provide info on all the changes made in this PR 💯 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

leonard33 committed:
edit book title
